pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -841,7 +841,20 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundTCPListenerOptsForPort
 	// in the map.
 	//
 	// Check if this TCP listener conflicts with an existing HTTP listener
-	if *currentListenerEntry, exists = listenerMap[*listenerMapKey]; exists {
+	*currentListenerEntry, exists = listenerMap[*listenerMapKey]
+
+	// If there is no listener with serviceListenAddress, check if a wildcard bind exists.
+	// This can happen if user specified a restricted service set on this port with wildcard
+	// bind and also adds a catchall('*/*' with no ports) listener. We should not create
+	// listener with serviceListenAddress in that case.
```

Comment:
I think I understand what this PR does, but I am not sure I understand why. What is wrong with creating a listener for

```
0.0.0.0_12000
<VIP>_12000
```
?

it seems like it should have no impact? is it just some perf optimization?

Reply:
Yes. In one of the usecase, we have 20 services on the same port but we want only one service with wildcard bind so we added sidecar and now if I add '*/*' at the end - it creates 20 additional listeners.

Reply:
I am a bit worried keeping the wildcard one will mean that we do not get customizations we may have otherwise gotten for the `<VIP>_12000` one

Reply:
Sure. No issues.